plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go

```diff
@@ -0,0 +1,276 @@
+package yandex_cloud_monitoring
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/selfstat"
+)
+
+// YandexCloudMonitoring allows publishing of metrics to the Yandex Cloud Monitoring custom metrics
+// service
+type YandexCloudMonitoring struct {
+	Timeout            internal.Duration `toml:"timeout"`
+	EndpointUrl        string            `toml:"endpoint_url"`
+	MetadataTokenUrl   string            `toml:"metadata_token_url"`
+	MetadataFolderUrl  string            `toml:"metadata_folder_url"`
+	Service            string            `toml:"service"`
+	FolderID           string            `toml:"folder_id"`
+	IAMTokenFromConfig string            `toml:"iam_token"`
+
+	IAMToken               string
```

Comment:
Nothing special is done with the IAMToken, it's just set if it's blank. you can merge these two.

Reply:
Thanks. Done